pkg/builder/backup_builder.go

```diff
@@ -77,7 +80,23 @@ func (b *BackupBuilder) ObjectMeta(opts ...ObjectMetaOpt) *BackupBuilder {
 
 // FromSchedule sets the Backup's spec and labels from the Schedule template
 func (b *BackupBuilder) FromSchedule(schedule *velerov1api.Schedule) *BackupBuilder {
-	labels := schedule.Labels
+	var labels = make(map[string]string)
+
+	// Check if there's explicit Labels defined in the Schedule object template
+	// and if present then copy it to the backup object.
+	if schedule.Spec.Template.Metadata.Labels != nil {
+		labels = schedule.Spec.Template.Metadata.Labels
+		logrus.WithFields(logrus.Fields{
+			"backup": fmt.Sprintf("%s/%s", b.object.GetNamespace(), b.object.GetName()),
+			"labels": schedule.Spec.Template.Metadata.Labels,
+		}).Debug("Schedule.template.metadata.labels set - using those labels instead of schedule.labels for backup object")
```

Comment:
@dsu-igeek Since the logger hasn't been initialized, I doubt that this is going to work. Hence could you please let me know how you would like this to be achieved/implemented?

Reply:
Perhaps the logger from https://github.com/vmware-tanzu/velero/blob/main/pkg/controller/schedule_controller.go#L260 could be passed down?

Reply:
@nrb I really don't think that it would work as the logger is available to any `controller object` but not to a `builder object` and hence there doesn't exist a logger to work with for the `BackupBuilder` object. Therefore I have initialized a new logger within the function scope for `FromSchedule` and have used that instead. Hope the approach makes sense to the team.